src/rpc/dial.rs

```diff
@@ -365,39 +388,189 @@ async fn get_auth_token(
 }
 
 impl DialBuilder<WithCredentials> {
-    /// attempts to establish a connection with credentials to the DialBuilder's given uri
-    pub async fn connect(self) -> Result<AddAuthorization<ViamChannel>> {
+    fn duplicate_uri(&self) -> Option<Parts> {
+        match &self.config.uri {
+            None => None,
+            Some(parts) => {
+                let uri = Uri::builder()
+                    .authority(parts.authority.clone()?)
+                    .path_and_query(parts.path_and_query.clone()?)
+                    .scheme(parts.scheme.clone()?);
+                Some(uri.build().ok()?.into_parts())
+            }
+        }
+    }
+
+    async fn get_mdns_uri(&self) -> Option<Parts> {
+        // CR erodkin: 10.1.12.11:36103
```

Comment:
@maximpertsov this is the host and port I was expecting to get when connecting to Khari's bot.